eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitiveImmutableKeySets.stg

```diff
@@ -30,7 +30,7 @@ import org.eclipse.collections.impl.set.mutable.primitive.<name>HashSet;
  *
  * @since 6.0.
  */
-class <className> extends AbstractImmutable<name>Set implements Serializable
+public class <className> extends AbstractImmutable<name>Set implements Serializable
```

Comment:
This should not need to be made public.

Reply:
The suggested generateCollisions approach works for all cases that come from out generated templates but there's a special hierarchy for *ByteSetTestCase that this didn't account for.

I've reverted these publics to package visible again and I've introduced the CollisionsUtil class to overcome the limitations for the *ByteSetTestCase hierarchy.